### ISSUE TITLE
Unnecessary calls to 'codeAction/resolve' on mouse hovers

### ISSUE THREAD

**Issue by dbalek** [microsoft/vscode#201871]
Type: <b>Bug</b>

Hovering with mouse over a list of available code actions that appears after clicking on a lightbulb displayed in the editor left margin, `codeAction/resolve` LSP call is invoked for each affected code action. This is not done when cycling the list using keyboard Up/Down when the `codeAction/resolve` is called only on the code action selected by user as to be performed.
The unnecessary calls to `codeAction/resolve` (coused by simple mouse hovers) represent a performace problem since the `codeAction/resolve` call is usually used to compute the `edit` property of a code action which could by quite costly (that was the reason to introduce the `codeAction/resolve` to LSP in version 3.16)

VS Code version: Code 1.85.1 (0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2, 2023-12-13T09:47:11.635Z)
OS version: Linux x64 6.6.3-1-default
Modes:

**Comment by justschen** [comment]
@dbalek yes this was intentional (ref https://github.com/microsoft/vscode/issues/175008) since we wanted to compute if there were edits in order to show a hover, which is why we resolve on hover (code action isn't applied, but just resolved in the background). 

I do see that this causes a 300ms long task to run on hover which isn't ideal, as you mentioned resolve is costly.

![Screenshot 2024-01-17 at 1 14 25 AM](https://github.com/microsoft/vscode/assets/54879025/506152dd-749d-4c63-8cf0-457f19c6f0ca)

giving this another look, thanks for bringing it up 👍

**Comment by stoichitescumaria** [comment]
I've got a similar problem as well. I am using the **`codeAction/resolve`** call to collect user input for the proposals that require it. With the new implementation, the user input will be requested whenever a proposal is hovered over, making it unselectable when using the mouse. Could you provide any additional information on the **`codeAction/resolve`** call as to whether it was triggered by selecting a proposal or by hovering over it?

**Comment by justschen** [comment]
Thanks for the report — this was a real performance issue. It was fixed in https://github.com/microsoft/vscode/commit/228f3ffc595 (PR #200765), which removed the unnecessary `action.resolve()` call from the hover handler. That fix shipped in VS Code 1.86.0.

On current `main`, the `onHover` delegate in the code action controller no longer triggers `codeAction/resolve` — it only checks the action kind and existing edit data to determine preview eligibility, so hovering over code action items is now lightweight.

Closing as completed. If you still see unexpected resolve calls on a current build, please open a new issue with details.
